tests/xmlrpcapi/miscellaneous_test.py

```diff
@@ -726,6 +726,309 @@ def test_xapi_object_edit(self, remote, token, remove_distro, create_kernel_init
+        create_distro(name_distro, "x86_64", "suse", path_kernel, path_initrd)
+        create_profile(name_profile, name_distro, "a=1 b=2 c=3 c=4 c=5 d e")
+        remote.xapi_object_edit(
+            "system",
+            name_system,
+            "add",
+            {
+                "name": name_system,
+                "profile": name_profile,
+            },
+            token,
+        )
+        remote.xapi_object_edit(
+            "system",
+            name_system,
+            "edit",
+            {
+                "name": name_system,
+                "interface": "eth2",
+            },
+            token,
+        )
+
+        # Act
+        result = remote.xapi_object_edit(
+            "system",
+            name_system,
+            "edit",
+            {
+                "name": name_system,
+                "mac_address": "aa:bb:cc:dd:ee:ff",
+            },
+            token,
+        )
+
+        # Assert
+        assert result
+        assert (
+            remote.get_system(name_system)
+            .get("interfaces", {})
+            .get("default", {})
+            .get("mac_address")
+            == "aa:bb:cc:dd:ee:ff"
+        )
+
+    def test_xapi_system_edit_two_interfaces_no_default_negative(
+        self,
+        remote,
+        token,
+        create_kernel_initrd,
+        create_distro,
+        create_profile,
+    ):
+        # Arrange
+        name_distro = "testsystem_xapi_edit"
+        name_profile = "testsystem_xapi_edit"
+        name_system = "testsystem_xapi_edit"
+        fk_kernel = "vmlinuz1"
+        fk_initrd = "initrd1.img"
+        basepath = create_kernel_initrd(fk_kernel, fk_initrd)
+        path_kernel = os.path.join(basepath, fk_kernel)
+        path_initrd = os.path.join(basepath, fk_initrd)
+        create_distro(name_distro, "x86_64", "suse", path_kernel, path_initrd)
+        create_profile(name_profile, name_distro, "a=1 b=2 c=3 c=4 c=5 d e")
+        remote.xapi_object_edit(
+            "system",
+            name_system,
+            "add",
+            {
+                "name": name_system,
+                "profile": name_profile,
+                "interface": "eth1",
+            },
+            token,
+        )
+        remote.xapi_object_edit(
+            "system",
+            name_system,
+            "edit",
+            {
+                "name": name_system,
+                "interface": "eth2",
+            },
+            token,
+        )
+
+        # Act & Assert
+        try:
+            remote.xapi_object_edit(
+                "system",
+                name_system,
+                "edit",
+                {
+                    "name": name_system,
+                    "mac_address": "aa:bb:cc:dd:ee:ff",
+                },
+                token,
+            )
+        except (ValueError):
+            assert True
+        else:
+            pytest.fail("Missing interface name did not raise an exception!")
```

Comment:
The pattern you used can be shortened to the following if I am not mistaken:

```suggestion
        with pytest.raises(ValueError):
            remote.xapi_object_edit(
                "system",
                name_system,
                "edit",
                {
                    "name": name_system,
                    "mac_address": "aa:bb:cc:dd:ee:ff",
                },
                token,
            )
```

Reply:
That's fine.
The new error message might be less clear, but it is easy to find out what is wrong by reading the code.
```
| E           Failed: DID NOT RAISE <class 'ValueError'>
```